src/cellarr/dataloader.py

```diff
@@ -1,15 +1,374 @@
+from collections import Counter
+import numpy as np
+import os
+import pytorch_lightning as pl
+from scipy.sparse import coo_matrix, csr_matrix
 import tiledb
+import torch
+from torch.utils.data import Dataset, DataLoader, WeightedRandomSampler
+from typing import List, Optional
+
+from cellarr.queryutils_tiledb_frame import subset_frame
 
 __author__ = "Tony Kuo"
 __copyright__ = "Jayaram Kancherla"
 __license__ = "MIT"
 
 # Turn off multithreading to allow multiple pytorch dataloader workers
 config = tiledb.Config()
-# config["sm.compute_concurrency_level"] = 1
-# config["sm.io_concurrency_level"] = 1
-# config["sm.num_async_threads"] = 1
-# config["sm.num_reader_threads"] = 1
-# config["sm.num_tbb_threads"] = 1
-# config["sm.num_writer_threads"] = 1
+config["sm.compute_concurrency_level"] = 1
+config["sm.io_concurrency_level"] = 1
+config["sm.num_async_threads"] = 1
+config["sm.num_reader_threads"] = 1
+config["sm.num_tbb_threads"] = 1
+config["sm.num_writer_threads"] = 1
 config["vfs.num_threads"] = 1
+
+
+class scDataset(Dataset):
```

Comment:
I would also add a `__repr__` and  `__str__` methods for readability

Reply:
Added